src/prefect/agent/kubernetes/agent.py

```diff
@@ -183,6 +188,60 @@ def manage_jobs(self) -> None:
                                         delete_job = True
                                         break
 
+                            # Report recent events for pending pods to flow run logs
+                            if pod.status.phase == "Pending":
+                                if not self.job_pod_event_timestamps.get(job_name):
+                                    self.job_pod_event_timestamps[job_name] = {}
+
+                                if not self.job_pod_event_timestamps[job_name].get(
+                                    pod_name
+                                ):
+                                    self.job_pod_event_timestamps[job_name][
+                                        pod_name
+                                    ] = datetime.min.replace(tzinfo=pytz.UTC)
+
+                                pod_events = self.core_client.list_namespaced_event(
+                                    namespace=self.namespace,
+                                    field_selector="involvedObject.name={}".format(
+                                        pod_name
+                                    ),
+                                    timeout_seconds=30,
+                                )
+
+                                for event in sorted(
+                                    pod_events.items, key=lambda x: x.last_timestamp
+                                ):
+                                    # Break out of loop if there are no new events
+                                    if (
+                                        event.last_timestamp
+                                        < self.job_pod_event_timestamps[job_name][
+                                            pod_name
+                                        ]
+                                    ):
+                                        break
```

Comment:
I think this check is incorrect - if we get a list of `[old_event, old_event, new_event]`, the check here will see the old event first and break out, dropping the chance to display the new event. I think you want to `continue` instead to skip displaying the event.

```suggestion
                                    # Skip old events
                                    if (
                                        event.last_timestamp
                                        < self.job_pod_event_timestamps[job_name][
                                            pod_name
                                        ]
                                    ):
                                        continue
```

Reply:
The loop sorts based on the timestamp and it will have the newest event first. The break prevents us from having to loop over everything in that case